common/rpc/interceptor/namespace_validator.go

```diff
@@ -212,6 +212,11 @@ func (ni *NamespaceValidatorInterceptor) StateValidationIntercept(
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
+
+	if !ni.shouldValidateNamespace(req) {
```

Comment:
I think I prefer the approach in https://github.com/temporalio/temporal/pull/5949 since it keeps the logic in the same place as special handling for other requests. You two should talk and decide on which way to go with

Reply:
I don't (the logic is basically "return nil if for some reason we don't what to validate namespace), but look like there are already many special cases like that in that code.